components/net/fetch/request.rs

```diff
@@ -233,17 +233,16 @@ fn fetch(request: Rc<RefCell<Request>>, cors_flag: bool) -> Response {
 }
 
 /// [Main fetch](https://fetch.spec.whatwg.org/#concept-main-fetch)
-fn main_fetch(request: Rc<RefCell<Request>>, _cors_flag: bool) -> Response {
+fn main_fetch(request: Rc<Request>, _cors_flag: bool) -> Response {
     // TODO: Implement main fetch spec
     let _ = basic_fetch(request);
     Response::network_error()
 }
 
 /// [Basic fetch](https://fetch.spec.whatwg.org#basic-fetch)
-fn basic_fetch(request: Rc<RefCell<Request>>) -> Response {
+fn basic_fetch(request: Rc<Request>) -> Response {
 
-    let req = request.borrow();
-    let url = req.url_list.last().unwrap();
+    let url = request.current_url().clone();
```

Comment:
There are other places where `current_url()` is followed by a `clone()`. Make sure you fix those as well.